ui/packages/catalog-ui-search/src/main/webapp/component/search-form/forms-sharing/search-form-sharing.collection.js

```diff
@@ -12,88 +12,92 @@
+        },
+    ],
+    addMySharedForms: function() {
+        templatePromise.then(() => {
+            if (!this.isDestroyed) {
                 $.each(sharedTemplates, (index, value) => {
                     if (this.checkIfShareable(value)) {
                         let utcSeconds = value.created / 1000;
                         let d = new Date(0);
                         d.setUTCSeconds(utcSeconds);
-                        this.addSearchForm(new SearchForm({
-                            createdOn: Common.getHumanReadableDate(d),
-                            id: value.id,
-                            name: value.title,
-                            type: 'custom',
-                            filterTemplate: value.filterTemplate,
-                            accessIndividuals: value.accessIndividuals,
-                            accessGroups: value.accessGroups,
-                            createdBy: value.creator
-                        }));
+                        this.addSearchForm(
+                            new SearchForm({
+                                createdOn: Common.getHumanReadableDate(d),
+                                id: value.id,
+                                name: value.title,
+                                type: 'custom',
+                                filterTemplate: value.filterTemplate,
+                                accessIndividuals: value.accessIndividuals,
+                                accessGroups: value.accessGroups,
+                                createdBy: value.creator,
+                            })
+                        );
                     }
                 });
                 this.doneLoading();
             }
-       });
-   },
-   checkIfShareable: function(template) {
-       if (this.checkIfInGroup(template) || this.checkIfInIndividiuals(template)) {
-           return true;
-       }
-       return false;
-   },
-   checkIfInGroup: function(template) {
-       let myGroups = user.get('user').get('roles');
-       let roleIntersection = myGroups.filter(function(n) {
-        return template.accessGroups.indexOf(n) !== -1;
-       });
+        });
+    },
+    checkIfShareable: function(template) {
+        return (
+            this.checkIfInGroup(template) ||
+            this.checkIfInIndividiuals(template)
+        );
+    },
+    checkIfInGroup: function(template) {
+        let myGroups = user.get('user').get('roles');
+        let roleIntersection = myGroups.filter(function(n) {
+            return template.accessGroups.indexOf(n) !== -1;
+        });
 
-       return !_.isEmpty(roleIntersection);
-   },
-   checkIfInIndividiuals: function(template) {
-       let myEmail = [user.get('user').get('email')];
-       let accessIndividualIntersection = myEmail.filter(function(n) {
-        return template.accessIndividuals.indexOf(n) !== -1;
-       });
+        return !_.isEmpty(roleIntersection);
+    },
+    checkIfInIndividiuals: function(template) {
```

Comment:
We aren't actually using these functions to determine whether or not the users are allowed to share/see metacards, right? I hope this is just for display. Any determination for sharing needs to be done on the back end against the PDP.

Reply:
@stustison Correct. I made a ticket for this earlier, currently in progress. Looking to piggy back the logic in the policy extension plugin, but haven't fully decided on a solution yet.